cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsCredentialConnector.java

```diff
@@ -0,0 +1,107 @@
+            return cloudCredentialStatus;
+        }
+        return importKeyPairs(authenticatedContext.getCloudCredential());
+    }
+
+    @Override
+    public CloudCredentialStatus delete(AuthenticatedContext authenticatedContext) {
+        AwsCredentialView awsCredential = new AwsCredentialView(authenticatedContext.getCloudCredential());
+        String keyPairName = awsCredential.getKeyPairName();
+
+        Collection<String> regions = platformParameters.regions().keySet();
+
+        for (String region : regions) {
+            if (!"CN_NORTH_1".equalsIgnoreCase(region) && !"GovCloud".equalsIgnoreCase(region)) {
+                try {
+                    AmazonEC2Client client = awsClient.createAccess(awsCredential, region);
+                    DeleteKeyPairRequest deleteKeyPairRequest = new DeleteKeyPairRequest(keyPairName);
+                    client.deleteKeyPair(deleteKeyPairRequest);
+                } catch (Exception e) {
+                    String errorMessage = String.format("Failed to delete public key [roleArn:'%s', region: '%s'], detailed message: %s",
+                            awsCredential.getRoleArn(), regions, e.getMessage());
+                    LOGGER.error(errorMessage, e);
+                }
+            }
+        }
+        return new CloudCredentialStatus(authenticatedContext.getCloudCredential(), CredentialStatus.DELETED);
+    }
+
+    private CloudCredentialStatus validateIamRoleIsAssumable(CloudCredential cloudCredential) {
+        AwsCredentialView awsCredential = new AwsCredentialView(cloudCredential);
+        try {
+            awsClient.retrieveCachedSessionCredentials(awsCredential);
+        } catch (AmazonClientException ae) {
+            if (ae.getMessage().contains("Unable to load AWS credentials")) {
+                String errorMessage =
+                        String.format("Unable to load AWS credentials: please make sure the deployer defined AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY");
+                LOGGER.error(errorMessage, ae);
+                return new CloudCredentialStatus(cloudCredential, CredentialStatus.FAILED, ae, errorMessage);
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Could not assume role '%s': check if the role exists and if it's created with the correct external ID",
+                    awsCredential.getRoleArn());
+            LOGGER.error(errorMessage, e);
+            return new CloudCredentialStatus(cloudCredential, CredentialStatus.FAILED, e, errorMessage);
+        }
+        return new CloudCredentialStatus(cloudCredential, CredentialStatus.CREATED);
+    }
+
+
+    private CloudCredentialStatus importKeyPairs(CloudCredential cloudCredential) {
+        AwsCredentialView awsCredential = new AwsCredentialView(cloudCredential);
+        try {
+            Collection<String> regions = platformParameters.regions().keySet();
+            for (String region : regions) {
```

Comment:
There is also an issue with OpenStack key import, and it also does not seem right that we import the key to every region, probably this CredentialConnector is conceptionally wrong and it needs to be invoked only before launchStack. We ned to talk about this